File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start = 0, end = height.length - 1;
        int ans = 0;

        while(start < end) {
            int minHeight = Math.min(height[start], height[end]);
            int len = end - start;
            ans = Math.max(ans, len * minHeight);
            if(height[start] > height[end]) end--;
            else start++;
        }

        return ans;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int start = 0, end = height.size() - 1;
        int ans = 0;

        while(start < end) {
            int h1 = height[start];
            int h2 = height[end];
            int minHeight = min(h1, h2);
            ans = max(ans, minHeight * (end - start));
            if(h1 > h2) end--;
            else start++;
        }

        return ans;
    }
};